### DISCUSSION TITLE
Mock 的翻譯

### DISCUSSION THREAD

**Discussion by ken71301** [python/python-docs-zh-tw#613] [Add Term]
### 初步確認

- [X] 我已經確認過 [術語列表](https://github.com/python/python-docs-zh-tw/wiki/%E8%A1%93%E8%AA%9E%E5%88%97%E8%A1%A8) 中沒有相關術語。
- [X] 我已經確認過 [Python 官方術語列表](https://docs.python.org/zh-tw/3/glossary.html) 中沒有相關術語。

### 術語原文

mock

### 術語翻譯

模擬 或 不翻譯

### 說明

在樂詞網的mock interview翻譯為模擬面試，因此暫定為「模擬」

但「模擬」也可能會與 simulate 的翻譯搞混，若想翻譯為「仿造」也可能與 fake 混淆

因此想徵求各位對這個詞翻譯的意見

感謝～

**Comment by 933yee** [comment]
我個人覺得「模仿」、「模擬」都可以

**Comment by mattwang44** [comment]
補充原 PR: #610

**Comment by griiid** [comment]
還沒有仔細推敲
但我覺得可以順帶的了解一下 unit test 中的 mock/stub/spy/fake 分別的差異

https://medium.com/@henry-chou/%E5%96%AE%E5%85%83%E6%B8%AC%E8%A9%A6%E4%B9%8B-mock-stub-spy-fake-%E5%82%BB%E5%82%BB%E6%90%9E%E4%B8%8D%E6%B8%85%E6%A5%9A-ba3dc4e86d86

我印象中 Python 的 unit test 好像只有 mock，但其實他可以做到的事情甚至有 spy (其他 stub 跟 fake 我還沒有用過/遇過所以不確定)

**Reply by mattwang44** [reply]
都可以做到，可能只是在不同的使用場景中會有不同的歸類

- mock object 只為了測試順暢而存在、沒有與被測試邏輯有互動時，就算是個 fake
- mock object 在測試的設定環節中有用到 `mock_obj.return_value.XXX = 'some value'` 時，就算是個 stub

**Comment by mattwang44** [comment]
my two cents ([ref](https://github.com/python/python-docs-zh-tw/pull/610#discussion_r1301966513))

>我自己認為意思比較接近「仿造」，和 simulation 相近但不完全一樣。不過既然 patch 不翻譯的話，mock 我覺得也可以不翻譯。
認真探討 simulation 與 mock 的差異的話，兩者都是 （廣義的）模擬，但前者是追求更完整的模擬、後者只為特定目的而達成最低需求的模擬；我自己是覺得找不到中文詞彙可以呈現這種差異。
測試領域的 stub, fake, spy, mock, dummy 在口語溝通上也通常直接使用英文原文而沒有慣用翻譯，即便我們自己定義了一套對映翻譯，應也難以讓人理解這幾個詞彙間的差異。

**Reply by griiid** [reply]
我看到這問題的第一時間也是覺得，好像在口語上是直接用英文溝通沒錯。
但不知道是不是翻譯上至少第一次出現需要後面括弧中文翻譯？
